diff --git a/arch/arm/boot/dts/am335x-boneblack.dts b/arch/arm/boot/dts/am335x-boneblack.dts
index 9312197316f0..25afd83e8195 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -12,6 +12,34 @@
 / {
 	model = "TI AM335x BeagleBone Black";
 	compatible = "ti,am335x-bone-black", "ti,am335x-bone", "ti,am33xx";
+
+	bone-gpio-devs {
+		compatible = "org,bone-gpio-sysfs";
+		gpio1 {
+			label = "gpio2.2";
+			bone-gpios = <&gpio2 2 GPIO_ACTIVE_HIGH>;
+		};
+		gpio2 {
+                        label = "gpio2.3";
+                        bone-gpios = <&gpio2 3 GPIO_ACTIVE_HIGH>;
+                };
+		gpio3 {
+                        label = "gpio2.4";
+                        bone-gpios = <&gpio2 4 GPIO_ACTIVE_HIGH>;
+                };
+		gpio4 {
+                        label = "gpio1.21";
+                        bone-gpios = <&gpio1 21 GPIO_ACTIVE_HIGH>;
+                };
+		gpio5 {
+                        label = "gpio1.22";
+                        bone-gpios = <&gpio1 22 GPIO_ACTIVE_HIGH>;
+                };
+		gpio6 {
+                        label = "gpio1.23";
+                        bone-gpios = <&gpio1 23 GPIO_ACTIVE_HIGH>;
+                };
+	};
 };
 
 &cpu0_opp_table {
